### PR TITLE
feat(customers): add delete and update

### PR DESCRIPTION
**Pull request by srindom**
Depends on #6137 

**What**
- Add update and delete methods to the customer module

### CODE REVIEW

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/customer/src/services/customer-module.ts [hunk 1]
```diff
@@ -109,6 +109,101 @@ export default class CustomerModuleService implements ICustomerModuleService {
...
+    data: Partial<CustomerTypes.CreateCustomerDTO>,
+    sharedContext?: Context
+  ): Promise<CustomerTypes.CustomerDTO[]>
+  update(
+    selector: CustomerTypes.FilterableCustomerProps,
+    data: Partial<CustomerTypes.CreateCustomerDTO>,
+    sharedContext?: Context
+  ): Promise<CustomerTypes.CustomerDTO[]>
+
+  @InjectManager("baseRepository_")
```
Comment (adrien2p) [inline packages/customer/src/services/customer-module.ts]: todo: InjectTransactionManager on mutation methods

File: packages/customer/src/services/customer-module.ts [hunk 2]
```diff
@@ -109,6 +109,101 @@ export default class CustomerModuleService implements ICustomerModuleService {
...
+        ...data,
+      }))
+    }
+
+    const customers = await this.customerService_.update(
+      updateData,
+      sharedContext
+    )
+
+    if (typeof idsOrSelector === "string") {
```
Comment (adrien2p) [inline packages/customer/src/services/customer-module.ts]: todo: isString

File: packages/customer/src/services/customer-module.ts [hunk 3]
```diff
@@ -109,6 +109,101 @@ export default class CustomerModuleService implements ICustomerModuleService {
...
+  }
+
+  delete(customerId: string, sharedContext?: Context): Promise<void>
+  delete(customerIds: string[], sharedContext?: Context): Promise<void>
+  delete(
+    selector: CustomerTypes.FilterableCustomerProps,
+    sharedContext?: Context
+  ): Promise<void>
+
+  async delete(
```
Comment (adrien2p) [inline packages/customer/src/services/customer-module.ts:172]: todo: missing transaction manager

File: packages/customer/src/services/customer-module.ts [hunk 4]
```diff
@@ -109,6 +109,101 @@ export default class CustomerModuleService implements ICustomerModuleService {
...
+    } else {
+      const ids = await this.customerService_.list(
+        idsOrSelector,
+        {
+          select: ["id"],
+        },
+        sharedContext
+      )
+      toDelete = ids.map(({ id }) => id)
+    }
```
Comment (adrien2p) [inline packages/customer/src/services/customer-module.ts]: in those context you can do the following

```suggestion
    let toDelete: string[] = Array.isArray(idsOrSelector) : idsOrSelector : [idsOrSelector as string]
  if (isObject(dsOrSelector)) {
      const ids = await this.customerService_.list(
        idsOrSelector,
        {
          select: ["id"],
        },
        sharedContext
      )
      toDelete = ids.map(({ id }) => id)
    }
```


wdyt?

**Review by adrien2p — APPROVED** [review]
LGTM 🔥 couple of todo and one that might be needed

**Pull request merged**